File: backend/src/main/java/com/back/back9/domain/user/dto/UserRegisterDto.java
package com.back.back9.domain.user.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record UserRegisterDto(
        @NotBlank
        String userLoginId,
        @NotBlank
        String username,
        @NotBlank
        String password,
        @NotBlank
        String confirmPassword
) {
    public UserRegisterDto {
        if (!password.equals(confirmPassword)) {
            throw new IllegalArgumentException("패스워드가 일치하지 않습니다");
        }
    }
}

File: backend/src/main/kotlin/com/back/back9/domain/user/dto/UserRegisterDto.kt
package com.back.back9.domain.user.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class UserRegisterDto(
    @field:NotBlank @field:Size(min = 2, max = 30)
    val userLoginId: String,
    @field:NotBlank @field:Size(min = 2, max = 30)
    val username: String,
    @field:NotBlank @field:Size(min = 2, max = 30)
    val password: String,
    @field:NotBlank @field:Size(min = 2, max = 30)
    val confirmPassword: String
) {
    init {
        require(password == confirmPassword) { "패스워드가 일치하지 않습니다" }
    }
}